app/questionnaire/navigation.py

```diff
@@ -31,86 +31,96 @@ def build_navigation(self, current_group_id, current_group_instance):
+            last_sect_group = self._get_last_sect_group(section, current_group_instance)
+            last_block = self._get_last_block_id(last_sect_group, current_group_instance)
 
-            last_block_location = Location(group['id'], 0, last_block_in_group['id'])
-            can_get_to_summary = is_summary_or_confirm_group and self.routing_path and \
-                last_block_location in self.routing_path
+            is_summary_or_confirm_group = last_block in ('summary', 'confirmation')
+            can_get_to_summary = is_summary_or_confirm_group and self.routing_path \
+                and Location(last_sect_group, 0, last_block) in self.routing_path
 
-            skip_group = self._should_skip_group(current_group_instance, group)
-            if not skip_group:
-                repeating_rule = SchemaHelper.get_repeat_rule(group)
+            if not self._should_skip_group(current_group_instance, first_sect_group):
+                repeating_rule = SchemaHelper.get_repeat_rule(SchemaHelper.get_group(self.survey_json, first_sect_group))
                 if repeating_rule:
-                    logger.debug("building repeating navigation", group_id=group['id'])
-                    navigation.extend(self._build_repeating_navigation(repeating_rule, group, current_group_id,
+                    navigation.extend(self._build_repeating_navigation(repeating_rule, section, current_group_id,
                                                                        current_group_instance))
-                elif last_group_id == group['id'] and can_get_to_summary:
-                    logger.debug("building navigation", group_id=group['id'])
+                elif last_group_id == first_sect_group and can_get_to_summary:
                     if len(self.completed_blocks) > 0 and set(self.routing_path[:-1]).issubset(self.completed_blocks):
-                        navigation.append(self._build_single_navigation(group, current_group_id, first_location))
-                elif last_group_id != group['id'] and not is_summary_or_confirm_group:
-                    logger.debug("building navigation", group_id=group['id'])
-                    navigation.append(self._build_single_navigation(group, current_group_id, first_location))
+                        navigation.append(self._build_single_navigation(section, current_group_id,
+                                                                        first_block_location, last_block))
+                elif last_group_id != first_sect_group and not is_summary_or_confirm_group:
+                    navigation.append(self._build_single_navigation(section, current_group_id,
+                                                                    first_block_location, last_block))
 
         return navigation
 
-    def _should_skip_group(self, current_group_instance, group):
-
-        skip_group = False
+    def _get_last_sect_group(self, section, current_group_instance):
```

Comment:
Yes he says sheepishly. Seems I lost a couple of changes during the conflict resolution. Done now.